powerbi-docs/developer/visuals/license-api.md

```diff
@@ -0,0 +1,188 @@
+The API also enables triggering the licensing related notifications that will appear on the Power BI visual and inform the user on the need to purchase the missing licenses.
+The visual should not display its own licensing UX, instead use one of Power BI predefined supported notifications as detailed below.  
+Learn more on Power BI Licensing and Translatability support [here](https://go.microsoft.com/fwlink/?linkid=2186097).
+
+>[!NOTE]
+>The `licensing` API is available from version 4.4. To find out which version you’re using, run the `pbiviz -V` command.
+
+
+## Retrieve visual's service plans that are assigned to the active user
+
+To get service plans assigned, add a call to `getAvailableServicePlans` (available via `IVisualLicenseManager`).    
+From performance perspective, attempt to fetch the licenses once, preferably in the `constructor` or the `init` calls, and save the result.  
+Once licenses are retrieved, they will be cached on Power BI host side during the Power BI session and any further calls to the same will return the cached data.
+
+```typescript
+export interface IVisualLicenseManager {
+        getAvailableServicePlans(): IPromise<powerbi.extensibility.visual.LicenseInfoResult>;
+    }
+```
+
+Retrieving the licenses might be a long operation, thus the `getAvailableServicePlans` call is an asynchronous call and should be handled as such in your code.  
+As a response to calling the method,  `LicenseInfoResult` object is returned.
+
+```typescript
+export interface LicenseInfoResult {
+        plans: ServicePlan[] | undefined;
+        isLicenseUnsupportedEnv: boolean; 
+        isLicenseInfoAvailable: boolean; 
+    }
+```
+
+* `plans` - an array of Service Plans purchased by the active user for this visual. A ServicePlan contains the service identifier and its state (ServicePlanState).  
+   Supported service plan states:
+
+| State | Description |
+| - | - |
+| Inactive | Indicates that the license is not active and shouldn't be used for provisioning benefits. |
+| Active | Indicates that the license is active and can be used for provisioning benefits. |
+| Warning | Indicates that the license is in grace period likely due to payment violation. |
+| Suspended | Indicates that the license is suspended likely due to payment violation. |
+| Unknown | Sentinel value. |
+
+
+* `isLicenseUnsupportedEnv` - indicates that the visual is being rendered in a Power BI environment that doesn't support licenses management or enforcement.   
+Currently, the following Power BI environments don't support license management or license enforcement:
+    *   Embedded; Publish To Web, PaaS embed (TBD: GA)
+    *   National clouds (Dependency on general support for translatability in national clouds)
+    *   RS (No planned support)
+    *   Exporting (PDF\PPT) using direct API calls (TBD: GA)
+    *   Rendering in Service Principal context (TBD: Public Preview)
```

Comment:
also - requires explanation: link to official docs

Reply:
N/A